src/DataStructures/Blaze/IntegerPow.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <blaze/math/DenseVector.h>
+#include <blaze/math/constraints/SIMDPack.h>
+#include <blaze/math/simd/BasicTypes.h>
+#include <blaze/system/Inline.h>
+#include <blaze/system/Vectorization.h>
+
+#include "Utilities/Math.hpp"
+
+namespace blaze {
+// This vectorized implementation of the integer pow function is necessary
+// because blaze does not offer its own version of a integer pow function.
+template <typename T>
+BLAZE_ALWAYS_INLINE SIMDdouble integer_pow(const SIMDf64<T>& v, const int e)
+#if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
+{
+  return _mm512_set_pd(::integer_pow((*v).eval().value[7], e),
```

Comment:
This sort of approach is generally risky (can't always avoid it) because this will actually not vectorize the code. It'll extract each element and then re-pack them. That's incredibly slow because it requires a ton of register loads and flushes. What I had in mind was:
```cpp
namespace detail {
BLAZE_ALWAYS_INLINE SIMDdouble integer_pow_impl(const SIMDf64<T>& x, const int e) {
  ASSERT(e >= 0, "Negative powers are not implemented");
  int ecount = e;
  int bitcount = 1;
  while (ecount >>= 1) {
    ++bitcount;
  }
  double result = 1.;
  while (bitcount) {
    result *= result;
    if ((e >> --bitcount) & 0x1) {
      result *= x;
    }
  }
  return result;
}
}  // namespace detail

BLAZE_ALWAYS_INLINE SIMDdouble integer_pow(const SIMDf64<T>& x, const int e) {
switch(e) {
    case 0:
      return blaze::set(1.0);
    case 1:
      return b;
    case 2:
      return b*b;
    case 3:
      return b*b*b;
    case 4:
      {
        const SIMDdouble b2 = b*b;
        return b2*b2;
      }
    case 5:
      {
        const SIMDdouble b2 = b*b;
        return b2*b2*b;
      }
    case 6:
      {
        const SIMDdouble b2 = b*b;
        return b2*b2*b2;
      }
    case 7:
      {
        const SIMDdouble b2 = b*b;
        return b2*b2*b2*b;
      }
    case 8:
      {
        const SIMDdouble b2 = b*b;
        const SIMDdouble b4 = b2*b2;
        return b4*b4;
      }
    default:
      return detail::integer_pow_impl(b, e);
    }
  }
```